### DISCUSSION TITLE
Zitadel integrate with tool that use group claim sush as kubelogin

### DISCUSSION THREAD

**Discussion by goupiN008** [zitadel/zitadel#4747] [Q&A] [answered]
Hi,
I've been trying to replace Keycloak with Zitadel as OIDC provider.
Tools that I integrated with Keycloak such as kubelogin, Hashicorp Vault, Kibana.
These tools require group claim (str or list of str) for config authorization  but I try to replace Keycloak group/role by using Zitadel role but I can't use Zitadel role claim because urn:zitadel:iam:org:project:roles is not type string or list of strings so I cant map group claim when I use Zitadel.

How can I resolve that?

**Comment by fforootd** [comment, marked as answer]
Hi There

You soon should be able to use ZITADELs role claim inside an action.
With this it will be possible to run a small js script to create the needed data format.

Last time I checked  @adlerhurst was working on an example and we track our effort in #3997
